src/algebra/ordered_ring.lean

```diff
@@ -216,6 +238,16 @@ multiplication with a positive number and addition are monotone. -/
 @[protect_proj]
 class ordered_comm_semiring (α : Type u) extends ordered_semiring α, comm_semiring α
 
+/-- Pullback an `ordered_comm_semiring` under an injective map. -/
+def function.injective.ordered_comm_semiring [ordered_comm_semiring α] {β : Type*}
+  [has_zero β] [has_one β] [has_add β] [has_mul β]
+  (f : β → α) (hf : function.injective f) (zero : f 0 = 0) (one : f 1 = 1)
+  (add : ∀ x y, f (x + y) = f x + f y) (mul : ∀ x y, f (x * y) = f x * f y) :
+  ordered_comm_semiring β :=
+-- I used `erw`, since `rw` did not work.
+{ mul_comm := λ a b, hf (by erw [mul, mul, mul_comm]),
```

Comment:
hf.comm_semiring should prove this for you

Reply:
```suggestion
{ ..hf.comm_semiring f zero one add mul,
```

Reply:
It did, thanks!